cirq/work/observable_measurement.py

```diff
@@ -0,0 +1,1317 @@
+# This file contains code modified from PyQuil.
```

Comment:
I think there's a lot happening in this single file. It would be much easier to read/extend/test the different components if we break this file into a module/submodule with each of the different component like `grouping_strategy`, `stopping_criteria` etc. in separate files.

Reply:
I can pull some of these things out